File: ballast-core/npm/src/domain/models/board-type.ts
export class BoardType {

    public static Rectangle: BoardType = new BoardType(0, "Rectangle", false);
    public static RegularPolygon: BoardType = new BoardType(1, "RegularPolygon", true );
    
    public readonly value: number;
    public readonly name: string;
    public readonly centerOrigin: boolean;

    private constructor(value: number, name: string, centerOrigin: boolean) {
        this.value = value;
        this.name = name;
        this.centerOrigin = centerOrigin;
    }

    public static list(): BoardType[] {
        return [
            BoardType.Rectangle,
            BoardType.RegularPolygon
        ];
    }

    public equals(BoardType: BoardType) {
        if (!BoardType) {
            return false;
        }
        return this.value == BoardType.value;
    }

    public static fromValue(value: number): BoardType {        
        let item = (value > -1) && BoardType.list().find(x => x.value == value);
        if (!item) {
            throw new Error(`Could derive board type from value (${value})`);
        }
        return item;
    }

    public static fromName(text: string): BoardType {           
        let item = !!text && BoardType.list().find(x => x.name.toLocaleLowerCase() == text.toLocaleLowerCase());
        if (!item) {
            throw new Error(`Could derive board type from text (${text})`);
        }
        return item;
    }

}

File: ballast-core/dotnet/src/domain/models/BoardType.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Ballast.Core.Domain.Models
{
    public class BoardType : StaticListTypeBase<BoardType>
    {

        public readonly static BoardType Rectangle = new BoardType(value: 0, name: nameof(Rectangle), centerOrigin: false);
        public readonly static BoardType RegularPolygon = new BoardType(value: 1, name: nameof(RegularPolygon), centerOrigin: true);

        public bool CenterOrigin { get; private set; }

        private BoardType(int value, string name, bool centerOrigin) : base(value, name)
        {
            CenterOrigin = centerOrigin;
        }

        public static IEnumerable<BoardType> List() => new[] {
            BoardType.Rectangle,
            BoardType.RegularPolygon
        };

        public static BoardType FromValue(int value) =>
            BoardType.List().Single(x => x.Value == value);

        public static BoardType FromName(string name) =>
            BoardType.List().Single(x => x.Name.ToLowerInvariant() == name.ToLowerInvariant());
            
    }
    
}